### PR TITLE
CloudFormation: create_change_set ChangeSetName more rules

### PR DESCRIPTION
**Pull request by tinngo365**
### Pull Request: Refactor Validation in `create_change_set`

**Summary:**

This PR refactors the `create_change_set` function to improve validation by centralizing it in a new `validate_create_change_set` function. This change ensures all parameters are validated consistently and makes it easier to extend validation in the future.

**Changes:**

- **Added `validate_create_change_set` Function**:
  - Centralizes validation for all parameters, currently including `change_set_name` rules (e.g., must start with a letter, be alphanumeric, and under 128 characters).

- **Updated `create_change_set`**:
  - Calls `validate_create_change_set` to validate inputs before processing.

**Testing:**

- Updated unit tests to cover validation scenarios.

[Documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudformation/client/create_change_set.html#:~:text=A%20change%20set%20name%20can%20contain%20only%20alphanumeric%2C%20case%20sensitive%20characters%2C%20and%20hyphens.%20It%20must%20start%20with%20an%20alphabetical%20character%20and%20can%E2%80%99t%20exceed%20128%20characters.)

image

### CODE REVIEW

File: tests/test_cloudformation/test_cloudformation_stack_crud_boto3.py [hunk 1]
```diff
@@ -1067,15 +1067,15 @@ def test_get_template_summary_for_stack_created_by_changeset_execution():
     conn.create_change_set(
         StackName="stack_from_changeset",
         TemplateBody=json.dumps(dummy_template3),
-        ChangeSetName="test_changeset",
```
Comment (tinngo365) [inline tests/test_cloudformation/test_cloudformation_stack_crud_boto3.py:1070]: Had to change this to follow the new rules

**Comment by tinngo365** [comment]
@bblommers could you explain what is wrong?

File: moto/cloudformation/utils.py [hunk 2]
```diff
@@ -134,3 +136,29 @@ def get_stack_from_s3_url(template_url: str, account_id: str, partition: str) ->
 
     key = s3_backends[account_id][partition].get_object(bucket_name, key_name)
     return key.value.decode("utf-8")  # type: ignore[union-attr]
+
+
+def validate_create_change_set(
+    stack_name: str,
```
Comment (bblommers) [inline moto/cloudformation/utils.py:142]: Can we change this here to only include the arguments that we're actually using?
The less arguments, the easier it is to read - and we can always change the signature later

File: moto/cloudformation/utils.py [hunk 3]
```diff
@@ -134,3 +136,29 @@ def get_stack_from_s3_url(template_url: str, account_id: str, partition: str) ->
...
+    change_set_name: str,
+    template: str,
+    parameters: Dict[str, str],
+    description: str,
+    change_set_type: str,
+    notification_arns: Optional[List[str]] = None,
+    tags: Optional[Dict[str, str]] = None,
+    role_arn: Optional[str] = None,
+) -> Any:
```
Comment (bblommers) [inline moto/cloudformation/utils.py:151]: Minor - but we can just simplify this to return `None`

File: tests/test_cloudformation/test_cloudformation_stack_crud_boto3.py [hunk 4]
```diff
@@ -2462,6 +2462,65 @@ def test_create_and_update_stack_with_unknown_resource():
             )
 
 
+@mock_aws
+def test_valid_change_set_name():
```
Comment (bblommers) [inline tests/test_cloudformation/test_cloudformation_stack_crud_boto3.py:2466]: I don't think this test is necessary? There are other tests that create a changeset, so they verify implicitly that the name is valid

**Review by bblommers — CHANGES_REQUESTED** [review]

**Comment by tinngo365** [comment]
Appreciate the feedback! Thank you

**Review by bblommers — APPROVED** [review]
LGTM - thank you for adding this to Moto @tinngo365!

**Pull request merged**